site/src/modules/notifications/NotificationsInbox/NotificationsInbox.tsx

```diff
@@ -75,6 +78,32 @@ export const NotificationsInbox: FC<NotificationsInboxProps> = ({
 		};
 	}, [updateNotificationsCache]);
 
+	const {
+		mutate: loadMoreNotifications,
+		isLoading: isLoadingMoreNotifications,
+	} = useMutation({
+		mutationFn: async () => {
+			if (!inboxRes) {
+				return;
+			}
+			const lastNotification =
+				inboxRes.notifications[inboxRes.notifications.length - 1];
+			const newRes = await fetchNotifications(lastNotification.id);
+			updateNotificationsCache((prev) => {
+				return {
+					unread_count: newRes.unread_count,
+					notifications: [...prev.notifications, ...newRes.notifications],
+				};
+			});
+		},
+		onError: (error) => {
+			displayError(
+				getErrorMessage(error, "Error loading more notifications"),
+				getErrorDetail(error),
+			);
+		},
+	});
```

Comment:
I just found way easier to use a mutation to update the data than use the `useInfiniteQuery` and its pages abstraction. Mostly when trying to update the previous query data.